File: Results/DER/Translation/translation_output/w_test/mistral/codenet/python/java/p02273.java
<s> [INST] Translate the following python code to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
1

Expected output:
0.00000000 0.00000000
33.33333333 0.00000000
50.00000000 28.86751346
66.66666667 0.00000000
100.00000000 0.00000000


```
import math


class Vector():
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def print(self):
        print(self.x, self.y)

    def __add__(self, o):
        return Vector(self.x + o.x, self.y + o.y)

    def __sub__(self, o):
        return Vector(self.x - o.x, self.y - o.y)

    def __mul__(self, o):
        return Vector(self.x * o, self.y * o)

    def __truediv__(self, o):
        return Vector(self.x / o, self.y / o)

    def rot(self, deg):
        rad = math.radians(deg)
        x = self.x * math.cos(rad) - self.y * math.sin(rad)
        y = self.x * math.sin(rad) + self.y * math.cos(rad)
        return Vector(x, y)


def koch(d, a, b):
    if d == 0:
        return

    s = a + (b-a)/3
    u = s + (s-a).rot(60)
    t = a + (b-a)*2/3

    koch(d-1, a, s)
    s.print()
    koch(d-1, s, u)
    u.print()
    koch(d-1, u, t)
    t.print()
    koch(d-1, t, b)


N = int(input())

a = Vector(0, 0)
b = Vector(100, 0)


a.print()
koch(N, a, b)
b.print()


```
[/INST]
```java
import java.util.Scanner;

class Vector {
    private double x, y;

    public Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public void print() {
        System.out.printf("%.12f, %.12f%n", x, y);
    }

    public Vector add(Vector o) {
        return new Vector(x + o.x, y + o.y);
    }

    public Vector sub(Vector o) {
        return new Vector(x - o.x, y - o.y);
    }

    public Vector mul(Vector o) {
        return new Vector(x * o.x, y * o.y);
    }

    public Vector truediv(Vector o) {
        return new Vector(x / o.x, y / o.y);
    }

    public Vector rot(double deg) {
        double rad = Math.toRadians(deg);
        double x = x * Math.cos(rad) - y * Math.sin(rad);
        double y = x * Math.sin(rad) + y * Math.cos(rad);
        return new Vector(x, y);
    }
}

public class KochSnowflake {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();

        Vector a = new Vector(0, 0);
        Vector b = new Vector(100, 0);

        a.print();
        koch(N, a, b);
        b.print();
    }

    public static void koch(int N, Vector a, Vector b) {
        if (N == 0) {
            return;
        }

        Vector s = a.add(b.sub(a));
        Vector u = s.add(s.sub(a).rot(Math.toRadians(60)));
        Vector t = a.add(b.sub(a).mul(2).div(3));

        koch(N - 1, a, s);
        s.print();
        koch(N - 1, s, u);
        u.print();
        koch(N - 1, u, t);
        t.print();
        koch(N - 1, t, b);
    }
}
```</s>

File: Results/DER/Translation/translation_output/w_test/mistral/codenet/java/python/p02273.py
<s> [INST] Translate the following java code to python and enclose your solution inside ```python```.
A sample test case is provided below:

Test input:
1

Expected output:
0.00000000 0.00000000
33.33333333 0.00000000
50.00000000 28.86751346
66.66666667 0.00000000
100.00000000 0.00000000


```
import java.util.Scanner;
import java.lang.Math;

class p02273 {

    public static class Tuple{
        public final double x;
        public final double y;
        public Tuple(double x, double y) {
            this.x = x;
            this.y = y;
        }
        public void print(){
            System.out.printf("%f %f\n", x, y);
        }
    }

    static void koch(int n, Tuple a, Tuple b){
        if (n == 0) return;

        Tuple s, t, u;
        double cos60 = Math.cos(Math.PI * 60.0 / 180.0);
        double sin60 = Math.sin(Math.PI * 60.0 / 180.0);
        s = new Tuple((2.0 * a.x + 1.0 *b.x) / 3.0, (2.0 * a.y + 1.0 * b.y) / 3.0);
        t = new Tuple((1.0 * a.x + 2.0 *b.x) / 3.0, (1.0 * a.y + 2.0 * b.y) / 3.0);
        u = new Tuple((t.x - s.x) * cos60 - (t.y - s.y) * sin60 + s.x,
                      (t.x - s.x) * sin60 + (t.y - s.y) * cos60 + s.y);

        koch(n - 1, a, s);
        s.print();
        koch(n - 1, s, u);
        u.print();
        koch(n - 1, u, t);
        t.print();
        koch(n - 1, t, b);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        Tuple a = new Tuple(0.0,0.0);
        Tuple b = new Tuple(100.0,0.0);

        a.print();
        koch(n,a,b);
        b.print();
    }
}


```
[/INST]
```python
from collections import namedtuple

class Tuple(namedtuple('Tuple', 'x, y')):
    def __new__(_cls, x=0.0, y=0.0):
        return super().__new__(_cls, x, y)

def koch(n, a, b):
    if n == 0:
        return

    s, t, u = (Tuple(2.0 * a.x + 1.0 * b.x) / 3.0,
                 Tuple(1.0 * a.x + 2.0 * b.x) / 3.0,
                 Tuple((t.x - s.x) * math.cos(math.pi * 60 / 180) - (t.y - s.y) * math.sin(math.pi * 60 / 180) + s.x,
                       (t.x - s.x) * math.sin(math.pi * 60 / 180) + (t.y - s.y) * math.cos(math.pi * 60 / 180) + s.y))

    koch(n - 1, a, s)
    s.print()
    koch(n - 1, s, u)
    u.print()
    koch(n - 1, u, t)
    t.print()
    koch(n - 1, t, b)

if __name__ == '__main__':
    n = int(input())
    a = Tuple(0.0, 0.0)
    b = Tuple(100.0, 0.0)

    a.print()
    koch(n, a, b)
    b.print()
```</s>
